main.js

```diff
@@ -119,48 +147,78 @@ function main() {
     Logger.log(matcher.allStudents[0].firstName);
     Logger.log(matcher.allStudents[0].preferences[0].name);
 
-    populateSheet(outputSheet, matcher);
+    populateSheet(outputSheet, dataSheet, matcher);
```

Comment:
`populateSheet` is getting long. Also, I can clearly see the part that uses `outputSheet` and the new part that uses `dataSheet`.
This function can be split into two functions, one to populate the output sheet and one to populate the data sheet. Both can be called here.

Reply:
Sounds good I'll separate them into two functions